source/utilities.cc

```diff
@@ -713,31 +713,40 @@ namespace aspect
 
           i++;
 
-          // TODO: add checks for coordinate ordering in data files
         }
 
       AssertThrow(i == (components + dim) * data_table.n_elements(),
                   ExcMessage (std::string("Number of read in points does not match number of expected points. File corrupted?")));
 
-      std_cxx11::array<unsigned int,dim> table_intervals;
-
+      // Fill the coordinate vectors with the coordinates in each direction
+      // Check that they are strictly ascending
       for (unsigned int i = 0; i < dim; i++)
         {
-          table_intervals[i] = table_points[i] - 1;
-
           TableIndices<dim> idx;
-          grid_extent[i].first = data_tables[i](idx);
-          idx[i] = table_points[i] - 1;
-          grid_extent[i].second = data_tables[i](idx);
+          double temp_coord = data_tables[i](idx);
+          double new_temp_coord = 0;
+          coordinate_values[i].push_back(temp_coord);
+          for (unsigned int n = 1; n < table_points[i]; n++)
+            {
+              idx[i] = n;
+              new_temp_coord = data_tables[i](idx);
+              AssertThrow(new_temp_coord > temp_coord,
+                          ExcMessage ("Coordinates in dimension "
+                                      + int_to_string(i)
+                                      + " are not strictly ascending. "));
+
+              coordinate_values[i].push_back(new_temp_coord);
+              temp_coord = new_temp_coord;
+            }
         }
 
+      // For each data component, set up a GridData
       for (unsigned int i = 0; i < components; i++)
         {
           if (data[i])
             delete data[i];
-          data[i] = new Functions::InterpolatedUniformGridData<dim> (grid_extent,
-                                                                     table_intervals,
-                                                                     data_tables[dim+i]);
+          data[i] = new Functions::InterpolatedTensorProductGridData<dim> (coordinate_values,
+                                                                           data_tables[dim+i]);
```

Comment:
Good question. I'll gladly leave that up to you. If you want to deduce it automatically, you'll need to have some tolerance. If you want to let the user specify it, you'll need to assert that the points really are equidistant.